packages/story-editor/src/components/library/panes/media/local/hotlink/hotlinkModal.js

```diff
@@ -0,0 +1,199 @@
+/*
+ * Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+/**
+ * External dependencies
+ */
+import { __, sprintf, translateToExclusiveList } from '@web-stories-wp/i18n';
+import { Input } from '@web-stories-wp/design-system';
+import { useState, useCallback } from 'react';
+import styled from 'styled-components';
+import {
+  createResource,
+  getFileNameFromUrl,
+  getFirstFrameOfVideo,
+  getImageDimensions,
+  getVideoDimensions,
+  hasVideoGotAudio,
+  getVideoLength,
+} from '@web-stories-wp/media';
+import PropTypes from 'prop-types';
+
+/**
+ * Internal dependencies
+ */
+import { useConfig } from '../../../../../../app';
+import Dialog from '../../../../../dialog';
+import useLibrary from '../../../../useLibrary';
+import {
+  useUploadVideoFrame,
+  getPosterName,
+} from '../../../../../../app/media/utils';
+import { isValidUrl, withProtocol } from '../../../../../../utils/url';
+
+const InputWrapper = styled.div`
+  margin: 16px 4px;
+  width: 470px;
+  height: 100px;
+`;
+
+function HotlinkModal({ isOpen, onClose }) {
+  const {
+    allowedFileTypes,
+    capabilities: { hasUploadMediaAction },
+  } = useConfig();
+  const { insertElement } = useLibrary((state) => ({
+    insertElement: state.actions.insertElement,
+  }));
+
+  const { uploadVideoPoster } = useUploadVideoFrame({});
+
+  const [errorMsg, setErrorMsg] = useState(false);
+
+  // @todo We're not really uploading anything here, so should we have a fixed list instead?
+  let description = __('No file types are currently supported.', 'web-stories');
+  if (allowedFileTypes.length) {
+    description = sprintf(
+      /* translators: %s is a list of allowed file extensions. */
+      __('You can insert %s.', 'web-stories'),
+      translateToExclusiveList(allowedFileTypes)
+    );
+  }
+  const error = sprintf(
+    /* translators: %s is the description with allowed file extensions. */
+    __('Invalid link. %s', 'web-stories'),
+    description
+  );
+  const [link, setLink] = useState('');
+
+  const getFileInfo = useCallback(
+    (value = link) => {
+      const ext = value.split(/[#?]/)[0].split('.').pop().trim();
```

Comment:
Would love to see this line as it's own function that lives in the media package. Feel like it would be useful elsewhere. Something like `getFileExt`. Maybe even have some unit tests. 

I have learn while working on media, not all files have exts, or if they do, the do weird things like, image.mp4.jpg. Having some unit tests here might be extremely helpful.

Reply:
This will be taken from the `HEAD` request in follow-up issues anyway and will be redundant.